Framework/DataHandling/inc/MantidDataHandling/LoadSpiceXML2DDet.h

```diff
@@ -106,7 +111,28 @@ class DLLExport LoadSpiceXML2DDet : public API::Algorithm {
   /// Get wavelength from workspace
   bool getHB3AWavelength(API::MatrixWorkspace_sptr dataws, double &wavelength);
 
+  /// Set output workspace's X-axs as lab-frame Q space
   void setXtoLabQ(API::MatrixWorkspace_sptr dataws, const double &wavelength);
+
+  /// SPICE detector XML file
+  std::string m_detXMLFileName;
+  /// XML node name in detector counts file
+  std::string m_detXMLNodeName;
+  /// Pixel size at X direction
+  size_t m_numPixelX;
```

Comment:
I'd tend to prefer having these variables as local variables in the relevant methods (and passing them around if they are required in some other places). If there's no need to cache values having a more restricted scope usually helps making the code easier to read. Most algorithm classes do not need to have member variables (or they need very few).